### ISSUE TITLE
[Feature Request]: Support custom make file projects and parse existing make files

### ISSUE THREAD

**Issue by srb623** [eranif/codelite#3201]
### Describe your idea in details

I develop my C++ applications using make files and this is currently not supported by codelite.  I know that there is cmake support but I do not want to have to re-write things just to use an IDE.  This is supported by Visual Studio Code but that IDE is very resource hungry and was interested in trying codelite out.  But as you do not support custom make file projects I cannot test codelite.  I do not want to have to enter include directories and other settings into codelite for each project when it is already contained in the make file.

**Comment by dghart** [comment]
Maybe I don't understand what you're looking for, but CodeLite _does_ support using your own makefiles. See e.g.:
![CustomMakefile](https://github.com/eranif/codelite/assets/6452587/406a3b10-0ea4-42d7-b388-87590bb79e8a)

**Comment by srb623** [comment]
@eranif and @dghart - thank you for your replies.  As suggested by @eranif I have created a workspace using the file system workspace type.  This allows me to use my own make files to clean, build, etc.  Once I read the documentation further I discovered the 'bear' command and now I have code completion, outlines etc.  All working well and now I can give codelite a good test to see whether it is less resource hungry than visual studio code.
